perf(oxlint): run e2e tests concurrently

camc314 opened the pull request:
Tests in the same file typically run in sequence. we can improve the performance by running these in parallel:

https://vitest.dev/guide/parallelism.html#test-parallelism

 - Before (sequential): ~5.2s total, e2e.test.ts ~5000ms
 - After (concurrent):  ~1.50s total, e2e.test.ts ~1200ms
    
~3.5x faster

camchenry reviewed (approved):
Makes sense to run concurrently with all of the async file/cmd operation here.

camc314 committed:
perf(oxlint): run e2e tests concurrently (#16651)

Tests in the same file typically run in sequence. we can improve the performance by running these in parallel:

https://vitest.dev/guide/parallelism.html#test-parallelism

 - Before (sequential): ~5.2s total, e2e.test.ts ~5000ms
 - After (concurrent):  ~1.50s total, e2e.test.ts ~1200ms

~3.5x faster
